tests/security/verify_fips_packages.pm

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019-2020 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: Compare package version with its expected version for SLE15SP3.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "basetest";
+use strict;
+use warnings;
+use version;
+use testapi;
+use utils qw(systemctl zypper_call);
+use Mojo::Util 'trim';
+
+my $final_result = 'ok';
+
+my %package = (
+    'kernel-default' => '5.14.21-150400.24.46.1',
+    'kernel-default-devel' => '5.14.21-150400.24.46.1',
+    'kernel-devel' => '5.14.21-150400.24.46.1',
+    'kernel-source' => '5.14.21-150400.24.46.1',
+    'kernel-default-devel-debuginfo' => '5.14.21-150400.24.46.1',
+    'kernel-default-debuginfo' => '5.14.21-150400.24.46.1',
+    'kernel-default-debugsource' => '5.14.21-150400.24.46.1',
+    'kernel-rt' => '5.14.21-150400.15.11.1',
+    'kernel-devel-rt' => '5.14.21-150400.15.11.1',
+    'kernel-source-rt' => '5.14.21-150400.15.11.1',
+    'libopenssl1_1' => '1.1.1l-150400.7.25.1',
+    'libopenssl1_1-hmac' => '1.1.1l-150400.7.25.1',
+    'libopenssl1_1-32bit' => '1.1.1l-150400.7.25.1',
+    'libopenssl1_1-hmac-32bit' => '1.1.1l-150400.7.25.1',
+    libgnutls3 => '3.7.3-150400.4.27.1',
+    'libgnutls30-hmac' => '3.7.3-150400.4.27.1',
+    'libgnutls-devel' => '3.7.3-150400.4.27.1',
+    libnettle => '3.7.3-150400.2.21',
+    libgcrypt20 => '1.9.4-150400.6.5.1',
+    'libgcrypt20-hmac' => '1.9.4-150400.6.5.1',
+    'libgcrypt-devel' => '1.9.4-150400.6.5.1',
+    'mozilla-nss-tools' => '3.79.4-150000.3.1',
+    'mozilla-nss-debugsource' => '3.79.4-150000.3.1',
+    'mozilla-nss' => '3.79.4-150000.3.1',
+    'mozilla-nss-certs' => '3.79.4-150000.3.1',
+    'mozilla-nss-devel' => '3.79.4-150000.3.1',
+    'mozilla-nss-debuginfo' => '3.79.4-150000.3.1',
+    libica4 => '4.2.1-150400.3.8.1',
+    'libica-tools' => '4.2.1-150400.3.8.1'
+);
+
+sub cmp_version {
+    my ($old, $new) = @_;
+    return $old eq $new;
+}
+
+sub cmp_packages {
+    my ($pcks, $pckv) = @_;
+    my $output = script_output("zypper se -xs $pcks | grep -w $pcks | head -1 | awk -F '|' '{print \$4}'", 100, proceed_on_failure => 1);
+    my $out = '';
+    for my $line (split(/\r?\n/, $output)) {
+        if (trim($line) =~ m/^\d+\.\d+(\.\d+)?/) {
+            $out = $line;
+            if (!cmp_version($pckv, $out)) {
+                $final_result = 'fail';
+                record_info("Package version", "The $pcks version is $out, but request is $pckv", result => $final_result);
+            }
+        }
+    }
+    if ($out eq '') {
+        record_info("Package version", "The $pcks package does not exist", result => 'fail');
+    }
+}
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+
+    foreach my $key (keys %package) {
+        my $pcks = cmp_packages($key, $package{$key});
```

Comment:
Can you produce what was expected vs what we got? to make the life of the reviewer easier...